### PR TITLE
doc: synchronise common functions with diseasy

### PR DESCRIPTION
**Pull request by RasmusSkytte**
### Intent
This PR intents to synchronise functionality between `diseasy` and [`diseasystore`](https://github.com/ssi-dk/diseasystore)

In addition, it does a few things:
* it fixes the documentation tests. Differences in folder structure between  `devtools::test()`. `devtools::check()`, and GitHub workflows meant that these tests were not run during `check()` and on GitHub
* It converts the non-ASCII test to a linter. The use of non-ASCII characters falls more into code-style issues than code-function issues and these are better handled with linters.  (The documentation tests could in principle also be run as linters, but the linting system would need a lot of "hacking" to get it to work, so leaving it as tests for now)

### Approach
N/A

### Known issues
N/A

### Checklist

* [x] The PR passes all local unit tests
* [ ] I have documented any new features introduced
* [ ] If the PR adds a new feature, please add an entry in `NEWS.md`
* [x] A reviewer is assigned to this PR

### CODE REVIEW

**Comment by RasmusSkytte** [comment]
Rebasing onto the updated main and removing some changes.

**Comment by RasmusSkytte** [comment]
Checks should succeed once this issue is fixed on `diseasystore` 
* [ ] https://github.com/ssi-dk/diseasystore/issues/60

File: R/0_documentation.R [hunk 1]
```diff
@@ -79,56 +79,78 @@ rd_target_schema <- function(type = "param") {
...
 }
 
 
-rd_end_date <- function(type = "param") {
+rd_slice_ts <- function(type = "param") {
   checkmate::assert_choice(type, c("param", "field"))
-  paste("end_date (`Date`)\\cr",
-        "Study period end.",
+  paste("(`Date` or `character`)\\cr",
+        "Date to slice the database on (used if source_conn is a database).",
```
Comment (marcusmunch) [inline R/0_documentation.R:99]: The function says "slice_ts", so I assume it's analogous to `SCDB::slice_ts`, which supports timestamps:

```suggestion
        "Date or timestamp (parseable by `as.POSIXct`) to slice the database on (used if source_conn is a database).",
```

File: R/0_documentation.R [hunk 2]
```diff
@@ -79,56 +79,78 @@ rd_target_schema <- function(type = "param") {
...
         ifelse(type == "field", " Read only.", ""))
 }
 
 
-rd_.data <- function(type = "param") { # nolint: object_name_linter
+rd_.data <- function(type = "param") {                                                                                  # nolint: object_name_linter
   checkmate::assert_choice(type, c("param", "field"))
   paste(".data\\cr",
-        "The data object to perform the operation on",
+        "The data object to perform the operation on.",
```
Comment (marcusmunch) [inline R/0_documentation.R:115]: A preposition is a bad word to end a sentence on:

```suggestion
        "The data object on which to perform the operation.",
```

**Review by marcusmunch — APPROVED** [review]

**Pull request merged**